src/Components/Blazor/WebAssembly.Authentication/src/Options/RemoteAuthenticationApplicationPathsOptions.cs

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.AspNetCore.Components.WebAssembly.Authentication
+{
+    /// <summary>
+    /// Represents the options for the paths used by the application for authentication operations. These paths are relative to the base.
+    /// </summary>
+    public class RemoteAuthenticationApplicationPathsOptions
+    {
+        /// <summary>
+        /// Gets or sets the path to the endpoint for registering new users.
+        /// </summary>
+        public string RegisterPath { get; set; } = RemoteAuthenticationDefaults.RegisterPath;
+
+        /// <summary>
+        /// Gets or sets the remote path to the remote endpoint for registering new users.
+        /// It might be absolute and point outside of the application.
+        /// </summary>
+        public string RemoteRegisterPath { get; set; }
+
+        /// <summary>
+        /// Gets or sets the path to the endpoint for modifying the settings for the user profile.
+        /// </summary>
+        public string ProfilePath { get; set; } = RemoteAuthenticationDefaults.ProfilePath;
+
+        /// <summary>
+        /// Gets or sets the path to the remote endpoint for modifying the settings for the user profile.
+        /// It might be absolute and point outside of the application.
+        /// </summary>
+        public string RemoteProfilePath { get; set; }
+
+        /// <summary>
+        /// Gets or sets the path to the login page.
+        /// </summary>
+        public string LoginPath { get; set; } = RemoteAuthenticationDefaults.LoginPath;
+
+        /// <summary>
+        /// Gets or sets the path to the login callback page.
+        /// </summary>
+        public string LoginCallbackPath { get; set; } = RemoteAuthenticationDefaults.LoginCallbackPath;
```

Comment:
Allowing explicit configuration for `LoginCallbackPath`, `LoginFailedPath`, or anything else that is an "action" on the authentication manager, seems like an opportunity for confusion and misconfiguration.

If we required the convention that the URL pattern needs to be `/some/url/{action}`, then the authentication manager can infer all the other URLs automatically. It just needs to strip off the final segment (which must correspond to the current action) and then it can append any other action name.

Do you think we could simplify it in that way?

Reply:
Also based on how these options get used by the system, these options rely on the assumption that the URL doesn't include either querystring or hash components, which might not be obvious to some devs.

Reply:
OK, I'm convinced